resource-service/common/git.go

```diff
@@ -84,6 +88,28 @@ func getAuthMethod(gitContext common_models.GitContext) (transport.AuthMethod, e
 		return publicKey, nil
 
 	} else if gitContext.Credentials.Token != "" && strings.HasPrefix(gitContext.Credentials.RemoteURI, "http") {
+		if gitContext.Credentials.GitProxyURL != "" {
+			customClient := &nethttp.Client{
+				Transport: &nethttp.Transport{
+					TLSClientConfig: &tls.Config{InsecureSkipVerify: gitContext.Credentials.GitProxyInsecure},
+					Proxy: nethttp.ProxyURL(&url.URL{
+						Scheme: gitContext.Credentials.GitProxyScheme,
+						User:   url.UserPassword(gitContext.Credentials.GitProxyUser, gitContext.Credentials.GitProxyPassword),
+						Host:   gitContext.Credentials.GitProxyURL,
+					}),
+				},
+
+				// 15 second timeout
+				Timeout: 15 * time.Second,
+
+				// don't follow redirect
+				CheckRedirect: func(req *nethttp.Request, via []*nethttp.Request) error {
+					return nethttp.ErrUseLastResponse
+				},
+			}
+
+			client.InstallProtocol("https", http.NewClient(customClient))
```

Comment:
can you add a comment describing what `client.InstallProtocol()` does exactly and why it is needed? to me it seem like this will cause all https traffic through this proxy, which would mean that other projects would then also go through the proxy, or am I understanding it wrong?

Reply:
No, it actually uses https protocol to route all the traffic (it should work like that now as default), but when the user specifies, that he does not want to use a secure connection, he specifies --git-proxy-insecure and uses the https protocol without the SSL verification. I will add a comment there